internal/bson/bson_test.go

```diff
@@ -182,6 +196,19 @@ func fuzzBinary(f *testing.F, testCases []testCase, newFunc func() bsontype) {
 			require.NoError(t, err)
 			assert.Equal(t, expectedB, bw.Bytes(), "WriteTo results differ")
 		}
+
+		// Test that generated value can be marshaled for logging.
+		// Currently, that seems to be the best place to check it since generating values from BSON bytes is very easy.
+		{
+			// not a "real" type
+			if _, ok := v.(*CString); ok {
+				t.Skip()
+			}
+
+			mB, err := fjson.Marshal(fromBSON(v))
```

Comment:
@chilagrow that's what I was thinking of

Reply:
@AlekSi yeah I've been reading this PR :pray:  I see what you did with tests that used CString, your comments explains a lot to me thanks!